src/getRecords.ts

```diff
@@ -0,0 +1,99 @@
+// IDEA: getRecords handles cases of:
+// [x] getting all records from a table
+// [x] getting specific records from a table (single or multiple)
+// the difference is handled by either supplying a set of record id's or not
+// (and other params like filterByFormula etc)
+
+// also - can either call this once the airtable singleton has been set up (in which case "airtable" is passed to the function under the hood), or call it by itself and pass the api key explicitly
+
+import AirtableSDK from "airtable"
+
+import { RecordZ } from "./types/record"
+import { SchemaZodT } from "./types/schema"
+import { SelectQueryParamsT } from "./types/queryParams"
+
+export interface GetRecordsWithIds<T extends SchemaZodT> {
+	tableId: string
+	baseId: string
+	schema: T
+	recordIds: string[]
+}
+
+export interface GetRecordsWithQuery<T extends SchemaZodT> {
+	tableId: string
+	baseId: string
+	schema: T
+	query?: SelectQueryParamsT
+}
+
+export interface GetRecordsWithIdsAndAirtable<T extends SchemaZodT>
+	extends GetRecordsWithIds<T> {
+	airtable: AirtableSDK
+}
+
+export interface GetRecordsWithQueryAndAirtable<T extends SchemaZodT>
+	extends GetRecordsWithQuery<T> {
+	airtable: AirtableSDK
+}
+export interface GetRecordsWithIdsAndApiKey<T extends SchemaZodT>
+	extends GetRecordsWithIds<T> {
+	apiKey: string
+}
+
+export interface GetRecordsWithQueryAndApiKey<T extends SchemaZodT>
+	extends GetRecordsWithQuery<T> {
+	apiKey: string
+}
+
+//TODO Add jsdoc as the types are not clear
```

Comment:
Similar to my earlier iteration, I am noticing the inferred intellisense pop over for this function is now very hard to read.

![image](https://github.com/bamboocreativenz/zod-airtable/assets/1431991/a1a2658a-658f-4296-a93b-00eee9001d4c)

So we will need to write JSDoc comments on the functions to provide better documentation.

Reply:
That's interesting... I wouldn't expect it to be so intense with interfaces. I'm intrigued to look at how other library authors do this.